AmplifyPlugins/Storage/Tests/AWSS3StoragePluginTests/AWSS3StoragePluginGetPresignedUrlTests.swift

```diff
@@ -176,4 +176,41 @@ final class AWSS3StoragePluginGetPresignedUrlTests: XCTestCase {
             "getPreSignedURL(serviceKey:signingOperation:expires:) \(expectedServiceKey) getObject \(expectedExpires)"
         ])
     }
+
+    /// - Given: A key to a non-existent S3 object
+    /// - When: An attempt to generate a pre-signed URL is performed using the `validateObjectExistence` option
+    /// - Then: The plugin throws an AmplifyStorageError.notFound error
+    func testGetURLNonExistentKeyWithValidateObjectExistenceOption() async throws {
+        storageService.validateObjectExistenceHandler = { key in
+            throw StorageError.keyNotFound(key, "", "")
+        }
+        let nonExistentKey = UUID().uuidString
+        let options = StorageGetURLRequest.Options(
+            pluginOptions: AWSStorageGetURLOptions(
+                validateObjectExistence: true
+            )
+        )
+        do {
+            let url = try await systemUnderTest.getURL(key: nonExistentKey, options: options)
+            XCTFail("Expecting error but got url: \(url)")
+        } catch StorageError.keyNotFound(let key, _, _, _) {
+            XCTAssertTrue(key.contains(nonExistentKey), "nonExistentKey: \(nonExistentKey), key: \(key)")
+        }
+    }
+
+    /// - Given: A key to a non-existent S3 object
+    /// - When: An attempt to generate a pre-signed URL is performed without using the `validateObjectExistence` option
+    /// - Then: The plugin returns without an error
+    func testGetURLNonExistentKeyWithoutValidateObjectExistenceOption() async throws {
+        storageService.validateObjectExistenceHandler = { key in
+            throw StorageError.keyNotFound(key, "", "")
+        }
+        let nonExistentKey = UUID().uuidString
+        let options = StorageGetURLRequest.Options(
+            pluginOptions: AWSStorageGetURLOptions(
+                validateObjectExistence: false
+            )
+        )
+        _ = try await systemUnderTest.getURL(key: nonExistentKey, options: options)
```

Comment:
suggestion: should we add an `XCTFail()` in the `catch` section of `try await systemUnderTest.getURL(key: nonExistentKey, options: options)`

Reply:
Do you mean to also remove the `throws` from this method?